src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -388,48 +388,100 @@ public ContentItemDisplay GetEmptyByKey(Guid contentTypeKey, int parentId)
             // if your user type doesn't have access to the Settings section it would not get this property mapped
-            if (mapped.DocumentType != null)
-                mapped.DocumentType.Name = Services.TextService.UmbracoDictionaryTranslate(mapped.DocumentType.Name);
+            if (display.DocumentType != null)
+                display.DocumentType.Name = Services.TextService.UmbracoDictionaryTranslate(display.DocumentType.Name);
 
             //remove the listview app if it exists
-            mapped.ContentApps = mapped.ContentApps.Where(x => x.Alias != "umbListView").ToList();
+            display.ContentApps = display.ContentApps.Where(x => x.Alias != "umbListView").ToList();
+            return display;
+        }
 
-            return mapped;
+        private ContentItemDisplay GetEmpty(IContentType contentType, int parentId)
+        {
+            var emptyContent = Services.ContentService.Create("", parentId, contentType, Security.GetUserId().ResultOr(0));
+            var mapped = MapToDisplay(emptyContent);
+            return CleanContentItemDisplay(mapped);
         }
 
         /// <summary>
-        /// Gets a collection of empty content items for all document types.
+        /// Gets an empty <see cref="ContentItemDisplay"/> for each content type in the IEnumerable, all with the same parent ID
         /// </summary>
-        /// <param name="contentTypeKeys"></param>
+        /// <remarks>Will attempt to re-use the same permissions for every content as long as the path and user are the same</remarks>
+        /// <param name="contentTypes"></param>
         /// <param name="parentId"></param>
-        [OutgoingEditorModelEvent]
-        public IDictionary<Guid, ContentItemDisplay> GetEmptyByKeys([FromUri] Guid[] contentTypeKeys, [FromUri] int parentId)
+        /// <returns></returns>
+        private IEnumerable<ContentItemDisplay> GetEmpties(IEnumerable<IContentType> contentTypes, int parentId)
         {
-            var result = new Dictionary<Guid, ContentItemDisplay>();
-
-            using var scope = _scopeProvider.CreateScope(autoComplete: true);
-            var contentTypes = Services.ContentTypeService.GetAll(contentTypeKeys).ToList();
+            var result = new List<ContentItemDisplay>();
+            var userId = Security.GetUserId().ResultOr(0);
+            var currentUser = Security.CurrentUser;
+            // We know that if the ID is less than 0 the parent is null.
+            // Since this is called with parent ID it's safe to assume that the parent is the same for all the content types.
+            var parent = parentId > 0 ? Services.ContentService.GetById(parentId) : null;
+            EntityPermissionSet permissions = null;
 
             foreach (var contentType in contentTypes)
             {
-                if (contentType is null)
+                var emptyContent = Services.ContentService.Create("", parentId, contentType, userId);
+
+                if (emptyContent is null)
```

Comment:
I don't think the Create method would ever return a null value? It's a factory method so in theory should never return null